packages/ember-flight-icons/src/components/flight-icon.ts

```diff
@@ -6,17 +6,49 @@
 import Component from '@glimmer/component';
 import { guidFor } from '@ember/object/internals';
 import { assert } from '@ember/debug';
-
 import { iconNames } from '@hashicorp/flight-icons/svg';
 
-export default class FlightIconComponent extends Component {
-  constructor() {
-    super(...arguments);
+import type { IconName } from '@hashicorp/flight-icons/svg';
+
+export interface FlightIconSignature {
+  Args: {
```

Comment:
We [tend to follow](https://github.com/hashicorp/design-system/blob/main/wiki/TypeScript-Migration.md#typescript-migration) the same order as in the documentation, so it's easy to review and compare:
https://helios.hashicorp.design/icons/usage-guidelines?tab=code#component-api